lib/p5.fab.js

```diff
@@ -258,376 +269,463 @@ class Fab {
+        this.cameraOrientation.x,
+        this.cameraOrientation.y,
+        this.cameraOrientation.z
+      );
+    }
+
+    this.cameraPosition.x = this.camera.eyeX;
+    this.cameraPosition.y = this.camera.eyeY;
+    this.cameraPosition.z = this.camera.eyeZ;
+    this.cameraOrientation.x = this.camera.centerX;
+    this.cameraOrientation.y = this.camera.centerY;
+    this.cameraOrientation.z = this.camera.centerZ;
+  }
+
+  drawCartesianPrinter() {
+    orbitControl(2, 2, 0.1);
+
+    // draw print bed
+    translate(-this.maxX / 2, 0, -this.maxY / 2);
+    rotateY(PI);
+    scale(-1, 1);
+    push();
+    translate(this.maxX / 2, 0, this.maxY / 2);
+    rotateY(PI / 12);
+    rotateX(PI / 12);
+    fill(254, 249, 152);
+    push();
+    translate(0, 2.5, 0);
+    box(this.maxX + 1, 5, this.maxY + 1); // build plate
+    pop();
+
+    push();
+    noFill();
+    translate(0, -this.maxZ / 2 + 1, 0);
+    stroke(220, 50, 32);
+    box(this.maxX, this.maxZ, this.maxY); // work envolope
+    pop();
+
+    noFill();
+    stroke(0);
+    translate(-this.maxX / 2, 0, -this.maxY / 2);
+  }
+
+  drawDeltaPrinter() {
+    orbitControl(2, 2, 0.1);
+
+    // draw print bed
+    translate(-this.radius, 0, -this.radius);
+    rotateY(PI);
+    scale(-1, 1);
+    push();
+    translate(this.radius, 0, this.radius);
+    rotateY(PI / 12);
+    rotateX(PI / 12);
+    fill(254, 249, 152);
+    push();
+    translate(0, 2.5, 0);
+    cylinder(this.radius + 1, 5); // build plate
+    pop();
+
+    push();
+    noFill();
+    translate(0, -this.maxZ / 2 + 1, 0);
+    stroke(220, 50, 32);
+    box((2 * this.radius) / sqrt(2), this.maxZ, (2 * this.radius) / sqrt(2)); // work envolope
+    pop();
+
+    // not sure if needed
+    noFill();
+    stroke(0);
+  }
+
+  /*****
+   * G-Code Commands
+   */
+  autoHome() {
+    var cmd = "G28";
+    this.add(cmd);
+    this.add("G92 E0");
+
+    return cmd;
+  }
+
+  setTemps(tNozzle, tBed) {
+    var cmd = `M104 S${tNozzle}`; // set nozzle temp without waiting
+    this.add(cmd);
+
+    cmd = `M140 S${tBed}`; // set bed temp without waiting
+    this.add(cmd);
+
+    // now wait for both
+    cmd = `M109 S${tNozzle}`;
+    this.add(cmd);
+    cmd = `M190 S${tBed}`;
+    this.add(cmd);
+
+    return cmd;
+  }
+
+  setNozzleTemp(t) {
+    var cmd = `M109 S${t}`; // wait for temp
+    this.add(cmd);
+    return cmd;
+  }
+
+  setBedTemp(t) {
+    var cmd = `M190 S${t}`;
+    this.add(cmd);
+    return cmd;
+  }
+
+  setAbsolutePosition() {
+    var cmd = "G90";
+    this.add(cmd);
+  }
+
+  setRelativePosition() {
+    var cmd = "G91";
+    this.add(cmd);
+  }
+
+  setERelative() {
+    var cmd = "M83";
+    this.add(cmd);
+  }
+
+  fanOn() {
+    var cmd = "M106";
+    this.add(cmd);
+  }
+
+  fanOff() {
+    var cmd = "M107";
+    this.add(cmd);
+  }
+
+  pausePrint(t = null) {
+    var cmd = t ? `M1 S${t}` : "M1 S10 this is a pause";
+    this.commandStream.unshift(cmd);
+  }
+
+  configure(config) {
```

Comment:
this method is run on construction with the default "Ender3" values but can be consequently called directly to switch to a different printer.
All field names were kept the same for backward compatibality. optional `radius`, `coordinateSystem` were introduced